common/djangoapps/course_modes/models.py

```diff
@@ -692,3 +697,38 @@ class CourseModesArchive(models.Model):
     expiration_date = models.DateField(default=None, null=True, blank=True)
 
     expiration_datetime = models.DateTimeField(default=None, null=True, blank=True)
```

Comment:
To ensure that `expiration_datetime` and `expiration_datetime_is_explicit` are always in sync, it might be good to add a setter for this field:

```python
@expiration_datetime.setter
def expiration_datetime(self, new_datetime):
    self.expiration_datetime_is_explicit = True
    self._expiration_datetime = new_datetime
```
and then rename the current `expiration_datetime` field while retaining its column name in the DB.